deployments/cloudfoundry/tile/scripts/generate_osdf.py

```diff
@@ -0,0 +1,412 @@
+
+     10.4. Distributing Source Code Form that is Incompatible With Secondary Licenses
+     If You choose to distribute Source Code Form that is Incompatible With Secondary Licenses under the terms of this version of the License, the notice described in Exhibit B of this License must be attached.
+
+Exhibit A - Source Code Form License Notice 
+
+     This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+If it is not possible or desirable to put the notice in a particular file, then You may include the notice in a location (such as a LICENSE file in a relevant directory) where a recipient would be likely to look for such a notice.
+
+You may add additional accurate notices of copyright ownership.
+
+Exhibit B - "Incompatible With Secondary Licenses" Notice 
+
+     This Source Code Form is "Incompatible With Secondary Licenses", as defined by the Mozilla Public License, v. 2.0.
+"""
+
+VALID_LICENSES = {
+	APACHE_NAME_STR: APACHE_LICENSE,
+	SIMPLIFIED_BSD_NAME_STR: SIMPLIFIED_BSD_LICENSE,
+	NEW_BSD_NAME_STR: NEW_BSD_LICENSE,
+	ISC_NAME_STR: ISC_LICENSE,
+	MIT_NAME_STR: MIT_LICENSE,
+	MPL_NAME_STR: MPL_LICENSE,
+}
+
+
+LICENSES_OUTPUT_PATH="license_finder_output.csv"
+
+class DEP_INDEX(IntEnum):
+	NAME = 0
+	VERSION = 1
+	LICENSE = 2
+
+def license_parser(licenses):
+	# String that's comma separated
+	return licenses.split(",")[0].rstrip().replace("\"", "")
+
+def parse_dependencies():
+	result = subprocess.run(["license_finder", "report", "--format", "csv"], stdout=subprocess.PIPE)
+
+	for dep_line in io.StringIO(result.stdout.decode()):
+		sections = dep_line.split(",")
+
+		if len(sections) < 3:
+			continue
+
+		dep_name_ver = sections[DEP_INDEX.NAME]
+
+		version = sections[DEP_INDEX.VERSION].replace("\"", "")
+		if len(version) > 0:
+			dep_name_ver += "-" + version
+		else:
+			print("Found invalid ver for " + sections[DEP_INDEX.NAME])
+
+		dep_license = license_parser(sections[DEP_INDEX.LICENSE])
+
+		VALID_LICENSES[dep_license].add_dependency(dep_name_ver)
+
+def get_file_contents(file_path):
+	output = ""
+	with open(file_path, 'r') as file:
+		output = file.read()
+
+	return output
+
+def get_primary_output():
+	output = ""
+	for license_name in LICENSE_LIST:
+		license = VALID_LICENSES[license_name]
+
+		output += license.primary_header + "\n\n"
+		for dep in license.dependencies:
+			output += "\t>>> " + dep + "\n"
+
+		output += "\n\n\n"
+
+	return output
+
+def get_appendix_output():
+	output = APPENDIX_HEADER
+
+	for license_name in LICENSE_LIST:
+		license = VALID_LICENSES[license_name]
+
+		output += license.secondary_header + "\n\n"
+
+		for dep in license.dependencies:
+			output += ">>> " + dep + "\n\n\n"
+
+	return output
+
+def main():
+	otelcol_ver = "0.95.0"
```

Comment:
Should this version be parameterized instead of hard-coded?

Reply:
Good call, I've updated to use a passed in argument.